src/workspaces/SettingsModal/utils.ts

```diff
@@ -158,3 +163,36 @@ export const modifyFirstSoftDeleteSetting = (
 
   return _.concat(softDeleteSettings, otherSettings);
 };
+
+/**
+ * Modifies the requester pays setting in the workspace settings.
+ * If no such setting exists and requester pays is set to enabled, it will be created.
+ *
+ * Note that any other settings will be preserved but moved to the end of the array.
+ */
+export const modifyRequesterPaysSetting = (
+  originalSettings: WorkspaceSetting[],
+  enabled: boolean
+): WorkspaceSetting[] => {
+  // Clone original for testing purposes and to allow eventing only if there was a change.
+  const workspaceSettings = _.cloneDeep(originalSettings);
+
+  const requesterPaysSettings: RequesterPaysSetting[] = workspaceSettings.filter((setting: WorkspaceSetting) =>
+    isRequesterPaysSetting(setting)
+  ) as RequesterPaysSetting[];
+  const otherSettings: WorkspaceSetting[] = workspaceSettings.filter((setting) => !isRequesterPaysSetting(setting));
+
+  // If no RequesterPaysSetting existed and requester pays is set to disabled, do nothing
+  if (requesterPaysSettings.length === 0 && !enabled) {
+    return workspaceSettings;
+  }
+  return _.concat(
+    [
+      {
+        settingType: 'GcpBucketRequesterPays',
+        config: { enabled },
+      } as RequesterPaysSetting,
+    ],
+    otherSettings
+  );
```

Comment:
Clarifying the overall behavior (partly for my own benefit):
- Eventing only happens if the new setting is different from the old one
- However, the array of new settings given to Rawls will include the new setting regardless of whether it is the same as the old one...
- ...unless the old one doesn't exist (i.e. hasn't been set yet) and the new one is equivalent to the "default" setting of this type, in which case it will not be included in the new settings